### PR TITLE
new(BorderedList): Add BorderedList component.

### PR DESCRIPTION
**Pull request by stefhatcher**
to: @milesj @stefhatcher @hayes @alecklandgraf

## Description

Simple component to help add borders to a list. Default has no padding on the items. Can add `compact` or `spacious` padding with consistent values as `Row` spacing.

## Motivation and Context

Noticed this pattern cropping up a lot.

## Testing

- Added some tests
- Added some storybook examples

## Screenshots

Storybook 2019-04-22 15-01-09
Core : BorderedList - Items with ` compact ` padding   ⋅ Storybook 2019-04-22 15-01-29
Core : BorderedList - Items with ` spacious ` padding   ⋅ Storybook 2019-04-22 15-01-50

## Checklist

- My code follows the style guide of this project.
- I have updated or added documentation accordingly.
- I have read the CONTRIBUTING document.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -6431,8 +6431,16 @@
       "integrity": "sha512-W/Ppz34uUme3LmXWjMgFlYyGnbo1hd9JvA0LNQ4EmieqVjg2GPYbj3H6tcdP2QGPGWdRKUqZVbVKLNIFVs/HiA==",
       "requires": {
         "browserslist": "^4.5.1",
+        "core-js": "3.0.0",
```
Comment (stefhatcher) [inline package-lock.json]: @milesj i'm getting this package-lock.json diff after running `npm run clean:hard && npm install && npm run bootstrap && npm run build` and `npm run storybook:build`.

not sure why i'm getting something different from master?

Comment (milesj) [reply package-lock.json]: Bootstrap doesn't respect the lock file correctly, so just don't commit the lock changes.

Comment (stefhatcher) [reply package-lock.json]: aaaah k!

File: packages/core/src/components/BorderedList/Item.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import withStyles, { css, WithStylesProps } from '../../composers/withStyles';
+
+export type Props = {
+  /** Item content. */
+  children: NonNullable<React.ReactNode>;
+  /** Render with reduced vertical padding. */
+  compact?: boolean;
```
Comment (milesj) [inline packages/core/src/components/BorderedList/Item.tsx]: Let's add that prop type that these 2 can't be used at the same time.

File: packages/core/src/components/BorderedList/index.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import childrenWithComponentName from '../../prop-types/childrenWithComponentName';
+import withStyles, { css, WithStylesProps } from '../../composers/withStyles';
+import Item from './Item';
+
+export { Item };
+
+type Props = {
```
Comment (milesj) [inline packages/core/src/components/BorderedList/index.tsx]: export this.

Comment (stefhatcher) [reply packages/core/src/components/BorderedList/index.tsx]: oops

File: packages/core/src/components/BorderedList/Item.tsx (hunk 2 above)
Comment (stefhatcher) [reply packages/core/src/components/BorderedList/Item.tsx]: updated!

File: packages/core/src/components/BorderedList/index.tsx (hunk 3 above)
Comment (stefhatcher) [reply packages/core/src/components/BorderedList/index.tsx]: updated!

**Pull request merged**